lib/galaxy/tool_util/linters/inputs.py

```diff
@@ -326,6 +335,20 @@ def lint_inputs(tool_xml, lint_ctx):
             if len(set(select_options_values)) != len(select_options_values):
                 lint_ctx.error(f"Select parameter [{param_name}] has multiple options with the same value", node=param)
 
+        if param_type == "boolean":
+            truevalue = param_attrib.get("truevalue", "true")
+            falsevalue = param_attrib.get("falsevalue", "false")
+            problematic_booleans_allowed = profile < "23.1"
+            lint_level = lint_ctx.warn if problematic_booleans_allowed else lint_ctx.error
+            if truevalue == falsevalue:
+                lint_level(
+                    f"Boolean parameter [{param_name}] needs distinct 'truevalue' and 'falsevalue' values.", node=param
+                )
+            if truevalue.lower() == "false":
+                lint_level(f"Boolean parameter [{param_name}] has invalid truevalue [{truevalue}].", node=param)
+            if falsevalue.lower() == "true":
+                lint_level(f"Boolean parameter [{param_name}] has invalid falsevalue [{falsevalue}].", node=param)
+
```

Comment:
Can one still use `truevalue="no"`? If I'm not wrong then xsd would allow this.

Reply:
I assume so - is that a problem?